64bit-portability-issue

I'm trying to build ext with the latest zephir and I'm geting (for both 32/64bit version):

> [  158s] E: php5-phalcon 64bit-portability-issue /home/abuild/rpmbuild/BUILD/php5-phalcon-2.0.10/zephir-master/parser/parser/parser.c:9352

``` log
if (program_length < 2) {
    return SUCCESS;
}
```

@steffengy did you reproduce it? I have problem to bulid phalcon on opensuse:

On local:

```
/home/mruz/Dropbox/git/ice/framework/vendor/phalcon/zephir/parser/parser/parser.c: In function ‘xx_parse_program’:
/home/mruz/Dropbox/git/ice/framework/vendor/phalcon/zephir/parser/parser/parser.c:9235:17: warning: return makes pointer from integer without a cast [-Wint-conversion]
 #define SUCCESS 1
                 ^
/home/mruz/Dropbox/git/ice/framework/vendor/phalcon/zephir/parser/parser/parser.c:9352:10: note: in expansion of macro ‘SUCCESS’
   return SUCCESS;
```

``` c
#define SUCCESS 1
#define FAILURE 0
//...

/**
 * Parses a comment returning an intermediate array representation
 */
zval *xx_parse_program(char *program, unsigned int program_length, char *file_path) {

    //...
    if (program_length < 2) {
        return SUCCESS;
    }
```

it should be cast to the proper pointer type at least

--- a/parser/parser/base.c
+++ b/parser/parser/base.c
@@ -116,7 +116,7 @@ static void xx_scanner_error_msg(xx_parser_status *parser_status){
 /**
  * Parses a comment returning an intermediate array representation
  */
-zval *xx_parse_program(char *program, unsigned int program_length, char *file_path) {
+zval *xx_parse_program(char *program, size_t program_length, char *file_path) {
 
 	char *error;
 	xx_scanner_state *state;
@@ -128,12 +128,8 @@ zval *xx_parse_program(char *program, unsigned int program_length, char *file_pa
 	/**
 	 * Check if the program has any length
 	 */
-	if (!program_length) {
-		return FAILURE;
-	}
-
 	if (program_length < 2) {
-		return SUCCESS;
+		return NULL;
 	}
 
 	/**

--- a/parser/zephir_parser.c
+++ b/parser/zephir_parser.c
@@ -27,7 +27,7 @@
 #include "ext/standard/info.h"
 #include "php_zephir_parser.h"
 
-extern zval *xx_parse_program(char *program, unsigned int program_length, char *file_path);
+extern zval *xx_parse_program(char *program, size_t program_length, char *file_path);
 
 /* {{{ proto string zephir_parse_file(string arg)
    Return a string to confirm that the module is compiled in */

